code/obj/machinery/printing_press.dm

```diff
@@ -531,13 +629,46 @@
 			B.layer = src.layer + 0.1
 		TRANSFER_OR_DROP(src, B)
 		books_to_make--
-		ink_level -= 2
-		paper_amt -= 2
+		src.ink_level -= 2
+		src.paper_amt -= 2
 
-	is_running = FALSE
+	src.is_running = FALSE
 	UpdateIcon() //just in case?
 	src.visible_message("\The [src] finishes printing and shuts down.")
 
+/obj/machinery/printing_press/proc/make_newspapers()
+	src.is_running = TRUE
+	var/newspapers_to_print = src.newspaper_amount
+	while (newspapers_to_print)
+		if (src.paper_amt < 2 || src.ink_level < 1) // can we keep doin printing?
+			if (src.paper_amt < 2) // If we don't have enough paper to print...
+				src.visible_message("\The [src] runs out of paper and stops printing.")
+			if (src.ink_level < 1) // ...or enough ink
+				src.visible_message("\The [src] runs out of ink and stops printing.")
+			src.is_running = FALSE
+			UpdateIcon()
+			break
+
+		var/obj/item/paper/newspaper/rolled/NP = new
+		NP.desc = ""
+		if (src.newspaper_publisher)
+			NP.publisher = src.newspaper_publisher
+			NP.name = "[NP.publisher] newspaper"
+		if (src.newspaper_headline)
+			NP.headline = src.newspaper_headline
+		// it can auto generate headlines and publisher if left alone.
+		NP.desc = "Its from [NP.publisher]. Its headline reads: [NP.headline]"
+		NP.info = pick_smart_string("newspaper.txt", "article")
```

Comment:
So you can't actually set the contents? That seems weird

Reply:
i figured having yet another field would be a hassle... but I can add it in.